### DISCUSSION TITLE
Automatically select regions in scanned documents

### DISCUSSION THREAD

**Discussion by artyb5** [ImageMagick/ImageMagick#2512] [Help]
Hi
I've used IM a bit, but would appreciate ideas for this. I found some related posts, but not quite the same. I've noticed that our photocopier at works scans to pdf in quite a clever way, and doing something similar from home scanners would be handy, particularly at this time of homeworking. 
The photocopier produces pdfs with a white background (for white paper) and containing several smaller jpgs where there is content on the page. Overall these files have a good contrast (not grey background) and produce small file sizes by not rasterizing the whole area. I wondered if something similar could be replicated using IM?
Thanks,
A

**Comment by snibgo** [comment]
It would help if you showed a sample.

I suppose your photocopier scans the entire page into pixels, then post-processes the image, such as trimming white borders, and identifying significant white areas within the image.

Yes, IM can do that. Perhaps your photocopier uses IM internally.

However, IM can't build PDF files intelligently. That is, each PDF page that IM writes will be a wrapper around a single embedded raster image. If you want each PDF page to be a number of raster images plus white areas, perhaps plus (vector) text, you need some other tool for that.

**Reply by Jmuccigr** [reply]
I'll just add that I've experienced a whole range of output from photocopier/scanners. Some just embed a single image in a PDF. Some do the page in strips (annoying). A whole variety of different images types can be embedded as well.

So you really have to check the output of your scanner. You might be better off having it produce images instead of PDFs, for example.

**Comment by artyb5** [comment]
Thanks for replies. I don't have an example scan that I could share as it would need to be something without and addresses etc on it, and I'm working from home so can't scan another document for this purpose. I just thought it would be timely with more people working from home and using more basic scanners at home. 
It sounds as though IM can't do what the copier does, which I explained as well as I could. It doesn't turn the text into vector images (no OCR) but includes multiple jpg files in the pdf where there is content other than the plain background on the page. That produces a small file which emails and saves well etc.
Any ideas of other open source projects that might be worth looking at?
Ta,
A
